src/transformers/models/rt_detr/modeling_rt_detr.py

```diff
@@ -1669,12 +1705,8 @@ def unfreeze_backbone(self):
         for param in self.backbone.parameters():
             param.requires_grad_(True)
 
-    @lru_cache(maxsize=32)
-    def generate_anchors(self, spatial_shapes=None, grid_size=0.05):
-        # We always generate anchors in float32 to preserve equivalence between
-        # dynamic and static anchor inference
-        dtype = torch.float32
-
+    @conditional_lru_cache(maxsize=32)
+    def generate_anchors(self, spatial_shapes=None, grid_size=0.05, device="cpu", dtype=torch.float32):
```

Comment:
The question here is `lru_cache` correctly handles `torch.dtype` argument for caching?

Reply:
Seems like it does, however there was still a problem with conditional_lru_cache where the cached function would be recreated everytime, so there was no cache hit. It works now though, I do see the cache hits increasing when I print them with `print("generate_anchors cache hits: ", self.__getattribute__(f"_cached_{func.__name__}").cache_info().hits)` inside the conditional_lru_cache decorator.